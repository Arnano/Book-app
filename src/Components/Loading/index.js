import React from 'react';
import PropTypes from 'prop-types';
import StyledLoading from './style';

import Image from '../Image';

const Loading = ({ classes, children, icon }) =>
  <StyledLoading
    className={classes}
    icon={icon} >
    <Image src={icon} alt='Spinner' />
    {children}
  </StyledLoading>

Loading.propTypes = {
  classes: PropTypes.string,
  children: PropTypes.node,
  icon: PropTypes.string
}

export default Loading;
