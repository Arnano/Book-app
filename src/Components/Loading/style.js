import Styled from 'styled-components';

const StyledLoading = Styled.div `
  display: flex;
  flex-direction: row;
  width: 960px;
  height: 100px;
  margin: 0 auto;
`;

export default StyledLoading;
