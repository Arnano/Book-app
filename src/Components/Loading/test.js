import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import spinner from '../../Style/assets/spinner.svg';

import Loading from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Loading', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loading icon={spinner}> Is loading </Loading>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Loading icon={spinner}> Is loading </Loading>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
