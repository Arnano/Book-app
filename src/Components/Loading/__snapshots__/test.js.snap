// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa fxDFpx"
>
  <img
    alt="Spinner"
    className="sc-bwzfXH iYNyHm"
    src="spinner.svg"
  />
   Is loading 
</div>
`;
